src/textual/widgets/_option_list.py

```diff
@@ -373,6 +373,7 @@ def add_options(self, new_options: Iterable[OptionListContent]) -> Self:
                 self._id_to_option[option._id] = option
             add_option(option)
         if self.is_mounted:
+            self.refresh(layout=True)
```

Comment:
Try `layout=self.styles.auto_dimensions` to avoid the layout if the widget wouldn't change dimensions.

Reply:
Done. Thanks for the suggestion.